src/napari_spatialdata/_viewer.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+from anndata import AnnData
+from loguru import logger
+from multiscale_spatial_image import MultiscaleSpatialImage
+from napari import Viewer
+from napari.layers import Labels, Points, Shapes
+from napari.utils.notifications import show_info
+
+from napari_spatialdata.utils._utils import _get_transform, _swap_coordinates, _transform_to_rgb
+
+if TYPE_CHECKING:
+    from napari.layers import Layer
+    from spatialdata import SpatialData
+
+
+class SpatialDataViewer:
+    def __init__(self, viewer: Viewer) -> None:
+        self.viewer = viewer
+        self.viewer.bind_key("Shift-L", self._inherit_metadata)
+
+    def _inherit_metadata(self, viewer: Viewer) -> None:
+        layers = list(viewer.layers.selection)
+        self.inherit_metadata(layers)
+
+    def inherit_metadata(self, layers: list[Layer]) -> None:
+        """
+        Inherit metadata from active layer.
+
+        A new layer that is added will inherit from the layer that is active when its added, ensuring proper association
+        with a spatialdata object and coordinate space.
+
+        Paramters
+        ---------
+        layers: list[Layer]
+            A list of napari layers of which only 1 should have a spatialdata object from which the other layers inherit
+            metadata.
+        """
+        # Layer.metadata.get would yield a default value which is not what we want.
+        sdatas = [layer.metadata["sdata"] for layer in layers if "sdata" in layer.metadata]
+        # len(sdatas) as opposed to 1 to ensure sdatas is not empty
+        sdata_count = sum(sdatas[0] is sdata for sdata in sdatas[1:]) if len(sdatas) != 1 else len(sdatas)
```

Comment:
> At the moment for some reason when calling the inherit function twice with the same selected layers, sdata_count becomes 2. I don't know exactly why this is the case and how to prevent it. It seems that assignment creates a copy rather than an extra reference to the same memory address. Any ideas?

I am sorry, but I am not sure I understand your test case. Is this what you did?

1. select 2 layers. one layer has SpatialData metadata, the other does not
2. you call the `inherit_metadata()` function 
3. With those same two layers selected, you call the `inherit_metadata()` function again
4. `sdata_count` is 2

If so, it seems like there are a couple of things wrong:

- line 47 to me suggests that sdata_count was meant to count the number of unique SpatialData objets. On the other hand, this count looks like it's counting the number of times the first spatial data object is the same as the other spatial data objects.
-  it doesn't seem to me like the count here should be 2 in the case above. it seems like it should be 1 since there would be 2 sdata objects and iterating starts at 1 (sdatas[1:]), the max value would be 1. I would check to make sure you're passing the layers you meant to into the `inherit_metadata()` function.

Reply:
I assumed that one might quickly select all layers in which case the behaviour is as follows assuming this test case:
1. Open a napari spatialdata with a SpatialData object.
2. Open 2 elements as layers, these are backed by the same Spatialdata object.
3. Open a new annotation layer, for example Shapes.
4. Select all layers and press `shift+L`.

Now the code does the following:
1. Get all sdata objects from the metadata in layers as list.
2. Check whether they all point to the same memory address -> sdata_count != 1 if no element in `sdatas` or more than 1 spatialdata memory address.
3. If check passes, get first layer with `sdata` in metadata.
4. Let annotation layers with no `sdata` in metadata inherit from the layer in step 3.

This still works fine up to this point. The shapes layer properly inherits from one of the layers that has `sdata`. Now when pressing `shift-L` again when the same layers are still selected we get the behaviour that `sdata_count` now is 2, which in my opinion shouldn't be the case. It seems that when inheriting the metadata a copy of `sdata` is created and thus there are now 2 memory addresses having a spatialdata object even though it should be one address.

The case for checking whether there are more than 2 addresses is because if it is the case than we don't necessarily know which spatialdata object to associate with the new annotation layer.

Reply:
> Hey @melonora , I am sorry, but I am having trouble reconciling your comment, the code, and your question above. Can you please write a test using the API (i.e., not with the keyboard shortcut) that covers this case so I can see where the current behavior is not meeting your expectations?

I will

Reply:
The main thing is that if we add sdata metadata to a layer from another layer, we should not have it that it seemingly points to 2 different spatialdata objects in memory, which it now seemingly does. Funnily enough when just opening multiple spatial elements backed by the same spatialdata object as layers in the viewer as you can see from the test then the count is 1.

Reply:
Also relocating this function so we directly use widget._sdata to see if it solves the problem is not an option as I think this likely will go away with a follow up PR regarding supporting multiple spatialdata objects.

Reply:
Just checked with id, all objects do have the same id so I will adjust the code to use id